### ISSUE TITLE
[Feature Request]Touch sensitivity modifier

### ISSUE THREAD

**Issue by Krusher97** [Arisotura/melonDS#499]
In Harvest Moon DS the only way to split a stack in your inventory is to "lightly" touch the item twice or lightly touch and drag it.

On real hardware its really finicky for me but it seems to be impossible to do in MelonDS as all clicks on the touchscreen with my mouse are "heavy" touches which select the whole stack. As far as I know this is the only game that even has this feature.

**Comment by Arisotura** [comment]
thing is, I'm not sure how I'd go about implementing this

like, the touchscreen is able to sense pressure somehow

also not sure whether DSi/3DS support it (they use different touchscreen hardware)

**Comment by Krusher97** [comment]
Having a hotkey that modifies the touchscreen pressure when clicking the mouse could work

Unless you mean the technical implementation in which case I am of no use.

**Comment by Arisotura** [comment]
no, I was talking about the user-facing part of this

and, your answer would be one way to go, but we'd need a way to visually indicate what the pressure is

dunno

**Comment by rzumer** [comment]
I'd go with a very fast (<0.5 second) ramp-up in pressure the longer the mouse button is held.

**Comment by poudink** [comment]
I feel like that'd just make things confusing. I think slider is the way to go.

**Comment by Arisotura** [comment]
relatedly, another thing to support somehow would be "multitouch"

with quotes because the DS doesn't actually support reporting multiple touch points. it just sorta averages all the touch points and returns that.

**Comment by Krusher97** [comment]
For multitouch you could have a virtual stylus which you could place on the screen like a pin which also gets clicked when you click your mouse.

**Comment by Arisotura** [comment]
what I had in mind was:
* right-click screen to place a new touch point, that stays pressed
* press left button for last touch point
* release left button when done, all touch points are cleared

**Comment by enhasiac** [comment]
old thread but I have some new info:

splitting stacks in Harvest Moon DS doesn't actually require a "light tap" like all the controls online make out, it requires a _fast_ tap, and I was able to split stacked items a few times (in MelonDS) after a few tries.

it's just that tapping that fast on a mouse is extremely inconsistent and uncomfortable and the game is still practically unplayable since the game is always on a timer and spending 15 seconds trying to split a stack every 2 minutes just isn't viable.

also able to verify that pressure sensitivity does nothing to the ability to split stacks since the increase/decrease pressure options in desume don't help either.

the solution people figured out in desume is to use slow motion to increase the input window of a fast tap, but MelonDS doesn't have that yet ¯\_(ツ)_/¯

~~pls add slow motion hotkey my fingers can't take this abuse~~

**Comment by Nixel1324** [comment]
TL;DR, "pressure" isn't a thing on the DS unless you count homebrew hacks.

I remember reading somewhere that the DS hardware doesn't actually have pressure sensitivity, and the DS firmware doesn't support it either. Certain homebrew apps mimic it by checking how many pixels register as "pressed", since pushing on the screen harder also pushes down more pixels. Apparently those apps are a pain to calibrate though, since not every DS is equally as sensitive. Which is probably why the DS firmware doesn't allow games to use that method.
